lib/client.js

```diff
@@ -1,68 +1,37 @@
 /*
- * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ * Copyright (c) 2018-2022 Digital Bazaar, Inc. All rights reserved.
  */
-'use strict';
+import * as retry from 'retry';
+import {config} from '@bedrock/core';
+import logger from './logger.js';
 
-const {config} = require('bedrock');
-const logger = require('./logger');
-const redis = require('promise-redis')();
-const Backoff = require('backo');
-
-module.exports = class Client {
+import {createClient} from 'redis';
+export class Client {
   constructor({
-    host = config.redis.host, port = config.redis.port,
-    prefix = config.redis.prefix
+    host = config.redis.host,
+    port = config.redis.port,
   } = {}) {
-    // TODO: allow backoff options to be passed in
-    const backoff = new Backoff({
-      min: config.redis.retry.min,
-      max: config.redis.retry.max,
-      jitter: config.redis.retry.jitter,
-      factor: config.redis.retry.factor
-    });
-    logger.verbose(`initializing client: ${host}:${port}`);
-    this.client = redis.createClient({
-      host,
-      port,
-      prefix,
-      retry_strategy: options => {
-        if(options.error && options.error.code === 'ECONNREFUSED') {
-          // end reconnecting on a specific error and flush all commands with an
-          // individual error
-          logger.error('connection refused');
-        }
-        if(options.total_retry_time > config.redis.retry.maxTotal) {
-          // end reconnecting after a specific timeout and flush all commands
-          // with a individual error
-          logger.error('exceeded max connection retry time', {
-            totalRetryTime: options.total_retry_time,
-            timesConnected: options.times_connected
-          });
-          // FIXME: improve error handling
-          return new Error('redis: exceeded max connection retry time');
-        }
-        if(options.times_connected > config.redis.retry.maxTimesConnected) {
-          // end reconnecting
-          logger.error('exceeded max connection times', {
-            totalRetryTime: options.total_retry_time,
-            timesConnected: options.times_connected
-          });
-          // FIXME: improve error handling
-          return new Error('redis: exceed max connection times');
+    logger.debug(`initializing client at: ${host}:${port}`);
+    this.client = createClient({
+      socket: {host, port, reconnectStrategy: attempt => {
+        if(attempt > config.redis.retryOptions.retries) {
```

Comment:
Can't suggest for the whole block here, but here's a start to bring it inline with our style:

```suggestion
      socket: {
        host,
        port,
        reconnectStrategy(attempt) {
          if(attempt > config.redis.retryOptions.retries) {
```

Reply:
Addressed here ee8ede4385fa401ec65af297227714b7043cb771